core/src/main/scala/squid/ir/PrettyPrinter.scala

```diff
@@ -0,0 +1,190 @@
+package squid
+package ir
+
+import utils._
+import squid.lang.Base
+import meta.RuntimeUniverseHelpers._
+import sru._
+
+class PrettyPrinter extends Base with TraceDebug {
+
+  override type Rep = Int => String
+  override type BoundVal = (String, TypeRep) // (name, type)
+  override type TypeRep  = TypSymbol
+  type TypSymbol = String
+  type MtdSymbol = String
+
+  def repEq(a: Rep, b: Rep): Boolean = a == b
+  def typLeq(a: TypeRep, b: TypeRep): Boolean = a == b
+
+  val indent = 2
+
+  private def removeBrackets(exp: String): String = exp.drop(1).dropRight(1)
+
+  val escape = (str: String) => str
+    .replace("\b", "\\b")
+    .replace("\n","\\n")
+    .replace("\t", "\\t")
+    .replace("\r","\\r")
+    .replace("\f", "\\f")
+    .replace("\"","\\\"")
+    .replace("\\","\\\\")
+
+  override def bindVal(name: String, typ: TypeRep, annots: List[Annot]): BoundVal = (name, typ)
+
+  override def readVal(v: BoundVal): Rep = offset => v._1 // get the name as string
+
+  override def const(value: Any): Rep = offset => value match {
+    case _: Char   => s"""'$value'"""
+    case v: String => s""""${escape(v)}""""
+    case _ => value.toString
+  }
+
+  override def lambda(params: List[BoundVal], body: => Rep): Rep = offset => {
+    val ps = params.map(p => s"${p._1}: ${p._2}") // map tp "name: Type"
+    val plist = ps.mkString(", ") // make list of all params
+    s"(($plist) => ${body(offset)})"
+  }
+
+  override def staticModule(fullName: String): Rep = offset => fullName
+
+  override def module(prefix: Rep, name: String, typ: TypeRep): Rep = offset => s"${prefix(offset)}.$name"
+
+  override def newObject(tp: TypeRep): Rep = offset => s"new $tp"
+
+  override def letin(bound: BoundVal, value: Rep, body: => Rep, bodyType: TypeRep): Rep = offset => {
+    val b = body(offset)
+    val v = value(offset)
+
+    if (bound._2.startsWith("squid.lib.Var")) {
+      val tpe = removeBrackets(bound._2.dropWhile(_ != '['))
+      " " * offset + s"var ${bound._1}: $tpe = $v\n$b"
+    } else {
+      " " * offset + s"val ${bound._1}: ${bound._2} = $v\n$b"
+    }
+  }
+
+  override def ascribe(self: Rep, typ: TypeRep): Rep = offset => s"(${self(offset)}: $typ)"
+
+  override def methodApp(self: Rep, mtd: MtdSymbol, targs: List[TypeRep], args: List[ArgList], tp: TypeRep): Rep = offset => {
+
+    val argumentLists = args.map(_.reps) // Seq[Int => String]
+    val arguments     = argumentLists.map(seq => seq.map(arg => arg(offset)))
+
+    val tArgList =  if (targs.nonEmpty) s"[${targs.mkString(", ")}]" else ""
+
+    val mtdDecoded = sru.TermName(mtd).decodedName
+
+    (self(offset), mtd) match {
+      case (squidLib, "IfThenElse") => {
```

Comment:
This should be `SquidLib`, where `val SquidLib = "squid.lib.package"` is defined somewhere in the `PrettyPrinter` class, no? Otherwise it's going to match any other method called `IfThenElse`.